pymc/distributions/discrete.py

```diff
@@ -1263,7 +1256,7 @@ class OrderedLogistic:
     def __new__(cls, name, eta, cutpoints, compute_p=True, **kwargs):
         p = cls.compute_p(eta, cutpoints)
         if compute_p:
-            p = pm.Deterministic(f"{name}_probs", p, dims=kwargs.get("dims"))
+            p = pm.Deterministic(f"{name}_probs", p)
```

Comment:
This is not trivial. p may have more or less dims than the respective Categorical variable, and these may or not be broadcasting.  I decided it was not worth the trouble for the time being. In general we need to think of an approach for automatic dims (see #6828)

Reply:
I think partial dim assignment is currently supported, so you could do something like `dims=(None,) * dim_diff + ('categories', )`, no? The last one at least is always known iiuc

Reply:
No this is not sufficient either because p could have a broadcastable dim on the same batch dimensions of the underlying categorical variable. `pm.Categorical(..., p=[[0.5, 0.5]], dims="obs”)`. You cannot say p has dims ("obs", None) because it has shape (1, 2) not (2, 2)

Reply:
Or you're suggestion adding a "categories dim" regardless of the rest. That has another problem, what if the user creates 2 variables with different number of categories? Then again you have two dims that are incompatible (same name, different shape). 

That can also be solved but I don't want to do it here, I just want to not break stuff for now

Reply:
I was assuming that "categories" already was provided somewhere, not that we generate or auto-name anything.

But yes I understand this is a rabbit hole, focusing on not breaking things is fine.

Reply:
In that case the PR is ready for review :)